packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -444,6 +405,14 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 	protected maybeUpdateChildState(child: SummarizerNodeWithGC, id: string) {
 		super.maybeUpdateChildState(child, id);
 
+		// If GC has run on this node and summarization isn't complete, this.wipSerializedUsedRoutes will be defined.
+		// In that case, update the used routes of the child node. This can happen in scenarios where a data store
+		// doesn't have any ops but its reference state changed. So, it gets realized during summarize after GC ran
+		// so GC would not have run on this node which is fine.
+		if (this.wipSerializedUsedRoutes !== undefined) {
+			child.updateUsedRoutes(this.childNodesUsedRoutes.get(id) ?? [""]);
```

Comment:
Why not do this all the time?

Reply:
We could. It would result in `updateUsedRoutes` being called multiple times in the regular case where a data store is realized during GC. It will be called here during child node creation and then the channel context would call it again after GC runs.

This is needed for a very specific case - A data store is not realized during GC (no ops) but realized during summarization (due to app code running or because it's used routes changed).